sdk/core/Azure.Core/src/Shared/Argument.cs

```diff
@@ -212,5 +212,24 @@ public static string CheckNotNullOrEmpty(string value, string name)
             AssertNotNullOrEmpty(value, name);
             return value;
         }
+
+        /// <summary>
+        /// Throws if <paramref name="value"/> is not null.
+        /// </summary>
+        /// <param name="value">The value to validate.</param>
+        /// <param name="name">The name of the parameter.</param>
+        /// <param name="message">The error message.</param>
+        /// <exception cref="ArgumentException"><paramref name="value"/> is not null.</exception>
+#if AZURE_NULLABLE
+        public static void AssertNull<T>([AllowNull] T value, string name, string message)
+#else
+        public static void AssertNull<T>(T value, string name, string message)
```

Comment:
@chunyu3, none of the other methods in the Argument class take a `message` -- would it be possible to follow this pattern and throw an exception like the others, for example:

```csharp
throw new ArgumentException("Value must be null.", name);
```

Reply:
Actually, I'm looking at the corresponding PR in autorest.csharp, and see that the different message is important: https://github.com/Azure/autorest.csharp/pull/1798/files#diff-b30ddd61987d52a5d783b4d351ce81a70dde80801008f4458b28c483923a466dR130

Would it be possible to make message nullable, and use "Value must be null" if there is no message passed to the method?

Reply:
None of the other methods need to because the message defaults to something useful. `ArgumentException` does not have a default message that is useful if you want to pass the `paramName` (recommended so the dev knows which param is an issue). If you pass `null` for the `message`, the default message is simply "(Parameter `paramName`)" which is not useful.